src/io/iomapserialize.cpp

```diff
@@ -250,22 +250,23 @@ void IOMapSerialize::saveTile(PropWriteStream &stream, const Tile* tile) {
 
 bool IOMapSerialize::loadHouseInfo() {
 	Database &db = Database::getInstance();
+	const uint8_t worldid = g_configManager().getNumber(WORLD_ID);
 
-	DBResult_ptr result = db.storeQuery("SELECT `id`, `owner`, `paid`, `warnings` FROM `houses`");
+	DBResult_ptr result = db.storeQuery("SELECT `xml_id`,`id`, `owner`, `paid`, `warnings` FROM `houses` where world_id =" + std::to_string(worldid));
 	if (!result) {
 		return false;
 	}
 
 	do {
-		House* house = g_game().map.houses.getHouse(result->getNumber<uint32_t>("id"));
+		House* house = g_game().map.houses.getHouse(result->getNumber<uint32_t>("xml_id"));
```

Comment:
Oh is this in reference to the ID in `houses.xml`? I would call the column something else, `xml` is an implementation detail. How about `map_id` to signal it's the house id within the map, as opposed to within a world.

Reply:
Changed to map_id.